src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -218,8 +218,10 @@ SimpleNameSyntax syntax
 				}
 
 				if( !immutabilityContext.IsImmutable(
-					type: argument,
-					kind: ImmutableTypeKind.Total,
+					new ImmutabilityQuery(
+						ImmutableTypeKind.Total,
+						argument
+					),

```

Comment:
These often went together.
This is the return type of that "GetStuff" function now.
I think introducing a named type makes things a bit clearer overall.